### PR TITLE
Store userinput

### LINKED ISSUE #63: Store user answer in database
We should have the possibility to store user answers to agent questions in our database.

- [ ] Update the database models so we can store answers (create a new model UserAnswers with a One-to-Many mapping from user to answers)
- [x] Store the answer to the question implemented in #60 in the database

Blocked by #60

### PR DESCRIPTION
**Pull request by bscheltinga**
Fixes #63 

To test, get the database running (migrate etc.), dump data: test_add_pa_evaluation_to_db.dump (https://github.com/PerfectFit-project/virtual-coach-db/tree/store-user-input-pa)

The pa_evaluation variable is changed based on the user input in the "action_save_pa_evaluation_to_db" action.

### CODE REVIEW

**Comment by bscheltinga** [comment]
One of the tasks in the issue was: "Update the database models so we can store answers (create a new model UserAnswers with a One-to-Many mapping from user to answers)"

This part is created as a new issue.

File: Rasa_Bot/requirements.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@
 rasa==2.8.0 # NB! when updating, make sure to also update:
             # * rasa-sdk base image in Dockerfile
             # * format of stories, domain.yml etc. if necessary
+git+https://github.com/PerfectFit-project/virtual-coach-db
```
Comment (svenvanderburg) [inline Rasa_Bot/requirements.txt:5]: This requirement is for the actions server right? So it should be in the `actions/requirements-actions.txt`.

File: Rasa_Bot/actions/actions.py [hunk 2]
```diff
@@ -135,3 +137,22 @@ async def run(self, dispatcher, tracker, domain):
...
+    def name(self):
+        return "action_save_pa_evaluation_to_db"
+
+    async def run(self, dispatcher, tracker, domain):
+
+        pa_evaluation_response = tracker.get_slot("pa_evaluation_response")
+
+        session = get_db_session()  # Creat session object to connect db
+
+        user_id = 1  # TODO_b take this value from database
```
Comment (svenvanderburg) [inline Rasa_Bot/actions/actions.py:154]: Can you create an issue for this TODO?

File: Rasa_Bot/actions/actions.py [hunk 3]
```diff
@@ -83,11 +85,11 @@ def _is_valid_input(value):
 
 
 # Have a custom response based on the pa_evaluation response
-class ActionPaEvaluationFormFilled(Action):
+class ActionUtterPaEvaluationFormFilled(Action):
     """Custom response based on PA evaluation form"""
 
     def name(self):
-        return "action_pa_likert_form_filled"
+        return "action_utter_pa_likert_form_filled"
```
Comment (svenvanderburg) [inline Rasa_Bot/actions/actions.py:92]: Shouldn't this name actually reflect the name of the class?
```suggestion
        return "action_utter_pa_evaluation_form_filled"
```

File: Rasa_Bot/actions/actions.py [hunk 4]
```diff
@@ -135,3 +137,22 @@ async def run(self, dispatcher, tracker, domain):
         )
 
         return [reminder]
+
+
+class ActionSavePaEvaluationToDB(Action):
```
Comment (svenvanderburg) [inline Rasa_Bot/actions/actions.py:142]:
```suggestion
class ActionStorePaEvaluation(Action):
```

File: Rasa_Bot/actions/actions.py [hunk 5]
```diff
@@ -135,3 +137,22 @@ async def run(self, dispatcher, tracker, domain):
...
+    """"To save user input from PA evaluation form to database"""
+
+    def name(self):
+        return "action_save_pa_evaluation_to_db"
```
Comment (svenvanderburg) [inline Rasa_Bot/actions/actions.py:146]:
```suggestion
        return "action_store_pa_evaluation"
```

File: Rasa_Bot/actions/actions.py [hunk 6]
```diff
@@ -135,3 +137,22 @@ async def run(self, dispatcher, tracker, domain):
...
+
+    async def run(self, dispatcher, tracker, domain):
+
+        pa_evaluation_response = tracker.get_slot("pa_evaluation_response")
+
+        session = get_db_session()  # Creat session object to connect db
+
+        user_id = 1  # TODO_b take this value from database
+        selected = session.query(Users).filter_by(nicedayuid=user_id).one()
+        selected.PA_evaluation = pa_evaluation_response
```
Comment (svenvanderburg) [inline Rasa_Bot/actions/actions.py:156]: Please update this based on my comments in https://github.com/PerfectFit-project/virtual-coach-db/pull/2

**Review by svenvanderburg — APPROVED** [review]
This is looking good! 

* This new action should be tested (and shouldn't be so difficult). Depending on your availability you can choose to add a test similar to https://github.com/PerfectFit-project/virtual-coach-server/blob/34c1bf5a168bd6e36006dec3534cb2e3f94f3c3a/Rasa_Bot/tests/test_actions.py#L15 or create an issue for it.
* Please resolve my comments in the code.

**Pull request merged**